features/call/impl/src/main/kotlin/io/element/android/features/call/impl/utils/ActiveCallManager.kt

```diff
@@ -173,6 +189,34 @@ class DefaultActiveCallManager @Inject constructor(
             )
         }
     }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private fun observeRingingCall() {
+        // This will observe ringing calls and ensure they're terminated if the room call is cancelled
+        activeCall
+            .filterNotNull()
+            .filter { it.callState is CallState.Ringing && it.callType is CallType.RoomCall }
+            .flatMapLatest { activeCall ->
+                val callType = activeCall.callType as CallType.RoomCall
+                // Get a flow of updated `hasRoomCall` values for the room
+                matrixClientProvider.getOrRestore(callType.sessionId).getOrNull()
+                    ?.getRoom(callType.roomId)
+                    ?.roomInfoFlow
+                    ?.map { it.hasRoomCall }
+                    ?: flowOf()
+            }
+            // We only want to check if the room active call status changes
+            .distinctUntilChanged()
+            // Skip the first one, we're not interested in it (if the check below passes, it had to be active anyway)
+            .drop(1)
```

Comment:
I am a bit scared about this `drop`. If it is not mandatory, can we just remove it?
Or maybe replace by
```kotlin
            // We only want to check if the room active call status changes
            .distinctUntilChanged()
            .filter { roomHasActiveCall -> !roomHasActiveCall }
            .onEach {
                // The call was cancelled
                incomingCallTimedOut()
            }
            .launchIn(coroutineScope)
```

(Unite test are still passing with this)

Reply:
IIRC it's not, because in a real life situation the ringing this flow will start before we get the room info, and by default the `roomHasActiveCall` will be `false` by default, instantly cancelling any incoming call as a result. We could replace it by some operator that takes pairs of events instead so we can compare `hadCall == true && newValue == false`, but that's technically doing exactly the same as the drop as it wouldn't trigger until there are 2 elements in the flow.

Reply:
ok